### PR TITLE
feat: add css for tabstripitem for bottom navigation android

### PR DESCRIPTION
**Pull request by MartoYankov**

### CODE REVIEW

**Comment by MartoYankov** [comment]
test package_version#rc

**Comment by MartoYankov** [comment]
test package_version#rc

File: e2e/ui-tests-app/app/bottom-navigation/background-color-page.css [hunk 1]
```diff
@@ -2,6 +2,18 @@ BottomNavigation {
   background-color: gold;
 }
 
+TabContentItem.special {
```
Comment (manoldonev) [inline e2e/ui-tests-app/app/bottom-navigation/background-color-page.css]: Can't get this to apply when running the ui-tests-app (Android).

File: e2e/ui-tests-app/app/bottom-navigation/text-transform-page.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+BottomNavigation {
+  text-transform: lowercase;
+}
+
+TabContentItem.special {
+  text-transform: uppercase;
+}
+
+TabStrip {
+  text-transform: capitalize;
```
Comment (manoldonev) [inline e2e/ui-tests-app/app/bottom-navigation/text-transform-page.css]: capitalize transform not applied when running the example on Android.

File: e2e/ui-tests-app/app/bottom-navigation/background-color-page.css (hunk 1 above)
Comment (manoldonev) [reply e2e/ui-tests-app/app/bottom-navigation/background-color-page.css]: Discussed this internally -- not implemented yet.

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 3]
```diff
@@ -103,6 +134,12 @@ function createTabItemSpec(tabStripItem: TabStripItem): org.nativescript.widgets
     const result = new org.nativescript.widgets.TabItemSpec();
     result.title = tabStripItem.title;
 
+    if (tabStripItem.backgroundColor) {
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: Outer if is redundant

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 4]
```diff
@@ -345,6 +382,7 @@ export class BottomNavigation extends TabNavigationBase {
         if (this.tabStrip && this.tabStrip.items) {
             const tabItems = new Array<org.nativescript.widgets.TabItemSpec>();
             this.tabStrip.items.forEach((item, i, arr) => {
+                (<any>item).index = i;
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: Let's at least define it on TabStripItem as `public index: number;` (like on TabContentItem)

File: tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts [hunk 5]
```diff
@@ -378,6 +416,80 @@ export class BottomNavigation extends TabNavigationBase {
...
+            this._bottomNavigationBar.setSelectedTabTextColor(value.android);
+        } else {
+            this._bottomNavigationBar.setTabTextColor(value);
+            this._bottomNavigationBar.setSelectedTabTextColor(value);
+        }
+    }
+
+    public setTabBarItemBackgroundColor(tabStripItem: TabStripItem, value: android.graphics.drawable.Drawable | Color): void {
+        // TODO: Should figure out a way to do it directly with the the nativeView
+        const tabStripItemIndex = this.tabStrip.items.indexOf(tabStripItem);
```
Comment (manoldonev) [inline tns-core-modules/ui/bottom-navigation/bottom-navigation.android.ts]: You are now setting index on tabStripItem (line 385) -- can't we use it instead?

**Comment by manoldonev** [comment]
test --ignore nested_frame_navigation#mdonev/tabs-bottom-nav-e2e api23 ios12

**Comment by manoldonev** [comment]
test package_version#rc

**Pull request merged**